agent/command/results_gotest.go

```diff
@@ -36,10 +38,20 @@ func (c *goTestResults) Name() string { return "gotest.parse_files" }
 // ParseParams reads the specified map of parameters into the goTestResults struct, and
 // validates that at least one file pattern is specified.
 func (c *goTestResults) ParseParams(params map[string]interface{}) error {
-	if err := mapstructure.Decode(params, c); err != nil {
+	var err error
+	if err = mapstructure.Decode(params, c); err != nil {
 		return errors.Wrapf(err, "error decoding '%s' params", c.Name())
 	}
 
+	if c.OptionalOutput != "" {
+		c.OptionalOutputBool, err = strconv.ParseBool(c.OptionalOutput)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	} else {
+		c.OptionalOutputBool = false
```

Comment:
You don't need this else case; the default value for boolean is already false

Reply:
done